convert2rhel/checks.py

```diff
@@ -448,3 +457,81 @@ def _bad_kernel_substring(kernel_release):
         )
         return True
     return False
+
+
+def check_package_updates():
+    """Ensure that the system packages installed are up-to-date."""
+    logger.task("Prepare: Checking if the packages are up-to-date")
+
+    # Get a total count of how many packages is left to update
+    total_packages = get_total_packages_to_update()
+
+    if total_packages > 0:
+        logger.warning(
+            "The system has %s packages not updated. "
+            "If you wish to update then before continuing with the conversion, execute the following step:\n\n"
+            "Run: yum update -y\n" % total_packages
+        )
+    else:
+        logger.info("System is up-to-date.")
+
+
+def is_loaded_kernel_latest():
+    """Check if the loaded kernel is behind or of the same version as in yum repos."""
+    logger.task("Prepare: Checking if the loaded kernel version is the most recent.")
+
+    # For OL/CentOS 8 the `kernel` is just a meta package, instead, we check for `kernel-core`.
+    # But for 6 and 7 releases, the correct way to check is using `kernel`.
+    package_to_check = "kernel-core" if system_info.version.major >= 8 else "kernel"
+
+    # The latest kernel version on repo
+    repoquery_output, _ = run_subprocess(
+        ["repoquery", "--quiet", "--qf", '"%{BUILDTIME}\\t%{VERSION}-%{RELEASE}"', package_to_check]
```

Comment:
Why do you use CLI instead of API. What is the targeted systems and in which versions? (RHEL8, 7, Fedora ...)

Reply:
Parsing outputs from DNF is sensitive to be contaminated by plugins.

Reply:
_get_packages_to_update_dnf() uses DNF API - therefore repoquery can be replaced by  dnf query-

```
query = base.sack.query().filterm(name=package_to_check)
# to detect running kernel
running_kernel = base.sack.get_running_kernel()

# Then you can try to compare version of packages by direct comparison of  package object. Be aware that running kernel can be not the latest one installed

```

Reply:
I think that using dnf API will allow to delete a lot of your code handling parsing output and comparison of versions.

Reply:
IIRC, by the time I developed this code I was not aware if `yum` had an similar API to interact with repoquery as dnf has, that's why I was using the `repoquery` for that stuff, as it was available in every distro we handle (RHEL8 and RHEL7).

I'm currently looking over the `yum` repository again to see if I can find any APIs that could be used here to have the same effect.